### PR TITLE
ListBox multiple item selection

### PR DESCRIPTION
**Pull request by Mylan719**
Currently in DotVVM there is no support for native HTML input `<select multiple>...</select>`
We do have `<dot:ListBox>` control but it does not support multiple selected items. 
ASP.NET Web Forms has `<ListBox SelectionMode="Multiple">` that can be used in this case.

I implemented  `SelectionMode` property for our `<dot:ListBox>`.

### CODE REVIEW

File: src/Framework/Framework/Controls/ListBox.cs [hunk 1]
```diff
@@ -32,16 +60,54 @@ protected override void AddAttributesToRender(IHtmlWriter writer, IDotvvmRequest
...
+            {
+                writer.AddKnockoutDataBind("value", this, SelectedValueProperty, renderEvenInServerRenderingMode: true);
+            }
+            writer.AddKnockoutDataBind("valueAllowUnset", "true");
         }
 
-        public static readonly DotvvmProperty SizeProperty =
-            DotvvmProperty.Register<int, ListBox>(t => t.Size, defaultValue: 10);
+        [ControlUsageValidator(Override = true)]
+        public static new IEnumerable<ControlUsageError> ValidateUsage(ResolvedControl control)
```
Comment (tomasherceg) [inline src/Framework/Framework/Controls/ListBox.cs:78]: Can you please add some tests for the usage validation? 
I am not sure what will happen if the SelectionMode=Multiple and the SelectedValues is not IEnumerable - the GetEnumerableType will probably throw an exception so this method will fail to.

**Review by tomasherceg — CHANGES_REQUESTED** [review]

File: src/Framework/Framework/Controls/ListBox.cs (hunk 1 above)
Comment (Mylan719) [reply src/Framework/Framework/Controls/ListBox.cs:78]: good catch you are right

Comment (Mylan719) [reply src/Framework/Framework/Controls/ListBox.cs:78]: Done ;)

**Comment by acizmarik** [comment]
We should reuse some parts of this implementation (ValidateUsage and tests for example).

**Pull request closed without merging**